test/integration/events.spec.js

```diff
@@ -168,6 +168,69 @@ describe('integration:events', () => {
+        .get('/events?query[dojoId]=95b351c7-8228-4a9d-8fcc-b6b8bf2dd0c2')
+        .set('Accept', 'text/calendar')
+        .expect('Content-Type', /text\/calendar/)
+        .expect(200);
+      const attributes = res.text.split('\r\n');
+      expect(attributes[0]).to.equal('BEGIN:VCALENDAR');
+      expect(attributes[1]).to.equal('VERSION:2.0');
+      expect(attributes[2]).to.equal('CALSCALE:GREGORIAN');
+      expect(attributes[3]).to.equal('PRODID:adamgibbons/ics');
+      expect(attributes[4]).to.equal('METHOD:PUBLISH');
+      expect(attributes[5]).to.equal('X-PUBLISHED-TTL:PT1H');
+      expect(attributes[6]).to.equal('BEGIN:VEVENT');
+      expect(attributes[7]).to.equal('UID:3ae8fc05-55b6-4ea1-ad85-4f385452f764');
+      expect(attributes[8]).to.equal('SUMMARY:Test event 3');
+      expect(attributes[9]).to.match(/DTSTAMP:[0-9]+T[0-9]+Z/);
+      expect(attributes[10]).to.match(/DTSTART:[0-9]+T[0-9]+Z/);
+      expect(attributes[11]).to.match(/DTEND:[0-9]+T[0-9]+Z/);
+      expect(attributes[12]).to.equal('URL:https://zen.coderdojo.com/events/3ae8fc05-55b6-4ea1-ad85-4f385452f764');
+      expect(attributes[13]).to.equal('ORGANIZER;CN=CoderDojo:mailto:info@coderdojo.com');
+      expect(attributes[14]).to.equal('END:VEVENT');
+      expect(attributes[15]).to.equal('BEGIN:VEVENT');
+      expect(attributes[16]).to.equal('UID:0e83d8e7-b991-4e4e-b3bd-36aa956f6754');
+      expect(attributes[17]).to.equal('SUMMARY:Test event 4');
+      expect(attributes[18]).to.match(/DTSTAMP:[0-9]+T[0-9]+Z/);
+      expect(attributes[19]).to.match(/DTSTART:[0-9]+T[0-9]+Z/);
+      expect(attributes[20]).to.match(/DTEND:[0-9]+T[0-9]+Z/);
+      expect(attributes[21]).to.equal('URL:https://zen.coderdojo.com/events/0e83d8e7-b991-4e4e-b3bd-36aa956f6754');
+      expect(attributes[22]).to.equal('ORGANIZER;CN=CoderDojo:mailto:info@coderdojo.com');
+
+      expect(attributes[23]).to.equal('END:VEVENT');
+      expect(attributes[24]).to.equal('BEGIN:VEVENT');
+      expect(attributes[25]).to.equal('UID:84c0310e-49ff-4607-99da-a5abb9fb5641');
+      expect(attributes[26]).to.equal('SUMMARY:Test event 5');
+      expect(attributes[27]).to.match(/DTSTAMP:[0-9]+T[0-9]+Z/);
+      expect(attributes[28]).to.match(/DTSTART:[0-9]+T[0-9]+Z/);
+      expect(attributes[29]).to.match(/DTEND:[0-9]+T[0-9]+Z/);
+      expect(attributes[30]).to.equal('URL:https://zen.coderdojo.com/events/84c0310e-49ff-4607-99da-a5abb9fb5641');
+      expect(attributes[31]).to.equal('ORGANIZER;CN=CoderDojo:mailto:info@coderdojo.com');
+      expect(attributes[32]).to.equal('END:VEVENT');
+      expect(attributes[33]).to.equal('BEGIN:VEVENT');
+      expect(attributes[34]).to.equal('UID:bcef18f8-b5ff-43a9-bc2c-7109f6e5dc20');
+      expect(attributes[35]).to.equal('SUMMARY:Test event 6');
+      expect(attributes[36]).to.match(/DTSTAMP:[0-9]+T[0-9]+Z/);
+      expect(attributes[37]).to.match(/DTSTART:[0-9]+T[0-9]+Z/);
+      expect(attributes[38]).to.match(/DTEND:[0-9]+T[0-9]+Z/);
```

Comment:
Hm. This is `attributes[NUMBER]` can become a problem in later stages of development when attributes are removed. This makes it harder to re-arrange them. Maybe there is something like

```
expect.any(attributes).to.match(/DTEND:[0-9]+T[0-9]+Z/);
expect.one(attributes).to.match(/DTEND:[0-9]+T[0-9]+Z/);
```

Reply:
Or better ^^

```
expect(attributes[i++]).to.match(/DTEND:[0-9]+T[0-9]+Z/);
```

Reply:
I agree it's quite messy once you start having some events. I like the incremental version as a solution.
I think i'll double that with a function to check a single event, and pass the offset based upon the number of properties being checked